gourmet/tests/test_allrecipes_com_plugin.py

```diff
@@ -1,29 +1,37 @@
 import os.path
 import unittest
-from bs4 import BeautifulSoup, BeautifulStoneSoup
+import urllib.request
+
+from bs4 import BeautifulSoup
 
 from gourmet.plugins.import_export.website_import_plugins import allrecipes_plugin
 
 
 class DummyImporter(object):
+
     class WebParser(object):
         pass
 
 
 class TestAllRecipesPlugin(unittest.TestCase):
+
     url = "https://allrecipes.com/recipe/asian-beef-with-snow-peas/"
 
-    def _read_html(self):
+    def _read_html(self, download=True):
+        if download:
+            with urllib.request.urlopen(self.url) as response:
+                self.text = response.read().decode("utf8")
+            return
+
         filename = os.path.join(os.path.dirname(__file__),
-                                'recipe_files',
-                                (os.path.splitext(os.path.basename(__file__))[0])[5:-7] + ".html")
-        with open(filename) as f:
+                                "recipe_files",
+                                "allrecipes_com.html")
+        with open(filename, encoding="utf8") as f:
             data = f.read()
-        # print(data)
         return data
 
     def setUp(self):
-        self.text = self._read_html()
+        self.text = self._read_html(False)
```

Comment:
If `download` is True, then None is returned and self.text is empty, even though `_read_html` had it set.

Reply:
Oops, will fix it.

Reply:
This issue will be gone, once `_read_html` is a `staticmethod`